### ISSUE TITLE
Review icon used for node reps

### ISSUE THREAD

**Issue by nchevobbe** [firefox-devtools/ux#108]
The icon currently used for node reps can be confusing.

![image](https://user-images.githubusercontent.com/578107/73169211-ea4b6500-40fb-11ea-899c-6f84a4f6221c.png)

in the console this icon acts as an element picker
in the ruleview and animation inspector, the icon acts as a highlighter

We should settle on one icon for picking elements and one icon for highlighting.

**Comment by fvsch** [comment]
I kinda like the current icon when used outside of Inspector, if the action is "go to Inspector and select that node". Especially since it's basically a smaller variant of the Inspector's tab icon.

So I would err on the side of figuring out a different icon for "highlight this element / all elements matching this selector". Although it's a tough concept to communicate visually.

Pinging @captainbrosset and @juliandescottes for the Inspector side.

**Comment by nchevobbe** [comment]
i wonder if we could have an icon version of the box model UI

<img width="196" alt="image" src="https://user-images.githubusercontent.com/578107/73270678-6b2f5d00-41df-11ea-9484-2e569b74eb54.png">

**Comment by captainbrosset** [comment]
It would be useful to make a comprehensive list of all the places where this icon is used and what it does in each. The hope is that some of these that act differently than most can be changed to be consistent with the rest.

**Comment by nchevobbe** [comment]
Here's the inventory (I think I reviewed all usages)

### console

<img width="694" alt="image" src="https://user-images.githubusercontent.com/578107/73370163-10b40080-42b4-11ea-8326-be68542194bf.png">

clicking on the icon jumps to the inspector and selects the node
hovering highlight the node on screen


### acessibility

<img width="340" alt="image" src="https://user-images.githubusercontent.com/578107/73370358-612b5e00-42b4-11ea-95b5-436a05056840.png">

clicking on the icon jumps to the inspector and selects the node
hovering highlight the node on screen
(same as console)

### dom panel

<img width="908" alt="image" src="https://user-images.githubusercontent.com/578107/73370517-a51e6300-42b4-11ea-9986-7718dfdf4217.png">

clicking on the icon jumps to the inspector and selects the node
hovering highlight the node on screen
(same as console)

### Rule view

<img width="291" alt="image" src="https://user-images.githubusercontent.com/578107/73370580-bbc4ba00-42b4-11ea-9362-6fe6a43439df.png">

clicking on the icon sets the highlighter for elements matching the selector (clicking again removes the highlighter)

### Layout view - grid section

<img width="315" alt="image" src="https://user-images.githubusercontent.com/578107/73370781-0b0aea80-42b5-11ea-8af3-e09b4884c207.png">

clicking on the icon jumps to the inspector and selects the node
hovering highlight the node on screen
(same as console)

### Layout view - Box model properties section (offset node)

<img width="421" alt="image" src="https://user-images.githubusercontent.com/578107/73370929-43aac400-42b5-11ea-8f2d-88bd4ad0a0b2.png">

clicking on the icon jumps to the inspector and selects the node
hovering highlight the node on screen
(same as console)

---

Basically, the rule view is at odd with all the other places we use this icon.

**Comment by fvsch** [comment]
Thanks a bunch Nicolas.
So we could try to change the icon for "Highlight all elements matching this selector" in Rules.

Do we know if end users struggle with this specific button? If they understand it as a way to jump to the Inspector, they may be ignoring it (since they're already in the Inspector).

Possible visual metaphors:
- highlighter pen https://thenounproject.com/search/?q=highlighter
- something that suggests the box model highlighter?
- something like a dotted selection box: https://thenounproject.com/search/?q=select&i=2407251
- ?

**Comment by captainbrosset** [comment]
Thanks a lot Nicolas.
There's one more case I know about in the Animation inspector:

<img width="474" alt="Screenshot 2020-01-30 at 14 25 55" src="https://user-images.githubusercontent.com/1152698/73453573-972c1900-436c-11ea-8ba3-90650ccecd6f.png">

Clicking on the icon highlights the node, clicking again turns off the highlighter.

So, finding a different icon for this one would be great.
As for the rule-view, that reminds me that we have a chrome-parity bug on file (I think) to actually highlight on selector hover instead of having an icon. So maybe we could do this instead and get rid of the icon.

**Comment by SebastianZ** [comment]
> As for the rule-view, that reminds me that we have a chrome-parity bug on file (I think) to actually highlight on selector hover instead of having an icon. So maybe we could do this instead and get rid of the icon.

I actually prefer the icon over the hover functionality, first because it is more obvious and second because it keeps the highlighter active and by that allows to scroll through the page to see all matches.

Sebastian

**Comment by violasong** [comment]
Idea:

1. Hover over element, property, font-name, etc (no icon needed): highlight in page and use light blue background-hover color. Current example: Markup, Fonts Panel. Future example: selector names, properties

2. In cases outside of Rules, hover to highlight, click to jump to first matching element in Inspector markup (not sure if we need the persist highlight feature in Rules.)
